tmail-backend/jmap/extensions-cassandra/src/main/scala/com/linagora/tmail/james/jmap/settings/CassandraJmapSettingsDAO.scala

```diff
@@ -0,0 +1,124 @@
+    .where(column(USER).isEqualTo(bindMarker(USER)))
+    .build())
+
+  private val selectAll: PreparedStatement = session.prepare(selectFrom(TABLE_NAME)
+    .all()
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  private val selectOne: PreparedStatement = session.prepare(selectFrom(TABLE_NAME)
+    .all()
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  private val deleteOne: PreparedStatement = session.prepare(deleteFrom(TABLE_NAME)
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  def insert(username: Username, jmapSettings: JmapSettings): SMono[JmapSettings] = {
+    val settings: Map[String, String] = jmapSettings.settings.map(item => (item._1.value.value, item._2.value))
+    val insertSetting = insert.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT)
+      .set(STATE, jmapSettings.state.value.toString, TypeCodecs.TEXT)
+      .set(SETTINGS, settings , MAP_OF_STRING_CODEC)
+
+    SMono.fromPublisher(executor.executeVoid(insertSetting)
+      .thenReturn(jmapSettings))
+  }
+
+  def selectAll(username: Username): SFlux[JmapSettings] =
+    SFlux.fromPublisher(executor.executeRows(selectAll.bind()
+      .set(USER, username.toString, TypeCodecs.TEXT))
+      .map(toJmapSettings))
+
+  def selectOne(username: Username): SMono[JmapSettings] =
+    SMono.fromPublisher(executor.executeSingleRow(selectOne.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT))
+      .map(toJmapSettings))
+
+  def selectState(username: Username): SMono[UuidState] =
+    SMono.fromPublisher(executor.executeSingleRow(selectOne.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT))
+      .map(toState))
+
+  def updateSetting(username: Username, newState: Option[UuidState], newSettings: Option[Map[JmapSettingsKey, JmapSettingsValue]]): SMono[Void] = {
+    val updateStatementBuilder: BoundStatementBuilder = updateStatement.boundStatementBuilder()
+    updateStatementBuilder.set(USER, username.asString(), TypeCodecs.TEXT)
+
+    newState match {
+      case Some(state) => updateStatementBuilder.set(STATE, state.toString, TypeCodecs.TEXT)
+      case None => updateStatementBuilder.unset(STATE)
+    }
+    val settings: Map[String, String] = newSettings.orNull.map(item => (item._1.value.value, item._2.value))
+
+    newSettings match {
+      case Some(setting) => updateStatementBuilder.set(SETTINGS, settings, MAP_OF_STRING_CODEC)
+    }
+
+    SMono.fromPublisher(executor.executeVoid(updateStatementBuilder.build()))
+  }
+
+  def deleteOne(username: Username): SMono[Void] =
+    SMono.fromPublisher(executor.executeVoid(deleteOne.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT)))
+
+  private def toJmapSettings(row: Row): JmapSettings = {
+    JmapSettings(settings = toSettings(row), state = UuidState(row.get(STATE, TypeCodecs.UUID)))
+  }
+
+  private def toSettings(row: Row): Map[JmapSettingsKey, JmapSettingsValue] =
```

Comment:
can you check it again?
It is "red" on my IDE
<img width="809" alt="image" src="https://github.com/linagora/tmail-backend/assets/81145350/1c61f9ab-0510-4d36-b02a-f68912d90580">